### PR TITLE
Ensure all log observers fork below ErrorFormattingObserver

### LINKED ISSUE #1673: Move CF observer to regular observer chain
The regular log observer and CF log observer should fork somewhere below the spec observer, if we want the same log and same CF ID in both CF and our own logs.

### PR DESCRIPTION
**Pull request by cyli**
So that both the CF observer and the regular observer benefit from spec.py, which should do splitting messages (done) and generating CF IDs (not done).

Fixes #1673.

Unfortunately, this global state, since we want a singleton fanout observer.

This combines `copying_wrapper` with `FanoutObserver`, since `FanoutObserver` is where the event forks. (also, this made things easier to test).

I've tested this branch on the dev vm against mimic and a requestb.in-like twisted service in place of cloudfeeds.

Messages seem to correctly go to both our logs and the request.bin service on both twisted==15.0.0 and twisted==15.3.0 .  For example, our logs (on twisted 15.3.0):

```
2015-08-25_00:06:48.56661 {
2015-08-25_00:06:48.56662   "@timestamp": "2015-08-25T00:06:48.564904", 
2015-08-25_00:06:48.56662   "@version": 1, 
2015-08-25_00:06:48.56663   "cloud_feed": true, 
2015-08-25_00:06:48.56663   "cloud_feed_id": "a8c8827c-55ae-47a5-b664-650cba38dfd6", 
2015-08-25_00:06:48.56663   "converger_run_id": "e6217b26-3bab-40f6-9195-400db2bb42eb", 
2015-08-25_00:06:48.56663   "format": "%(log_legacy)s", 
2015-08-25_00:06:48.56664   "host": "vm", 
2015-08-25_00:06:48.56664   "lb_id": "26924", 
2015-08-25_00:06:48.56664   "level": 6, 
2015-08-25_00:06:48.56665   "log_format": "{log_text}", 
2015-08-25_00:06:48.56665   "log_legacy": "<twisted.logger._stdlib.StringifiableFromEvent object at 0x4cdc150>", 
2015-08-25_00:06:48.56665   "log_level": "<LogLevel=info>", 
2015-08-25_00:06:48.56665   "log_namespace": "log_legacy", 
2015-08-25_00:06:48.56666   "log_system": "otter", 
2015-08-25_00:06:48.56666   "log_text": "convergence-remove-clb-nodes", 
2015-08-25_00:06:48.56666   "log_time": 1440461208.564904, 
2015-08-25_00:06:48.56667   "message": "Removing nodes from CLB 26924: ['144480', '852956', '972535']", 
2015-08-25_00:06:48.56667   "nodes": [
2015-08-25_00:06:48.56668     "144480", 
2015-08-25_00:06:48.56668     "852956", 
2015-08-25_00:06:48.56668     "972535"
2015-08-25_00:06:48.56668   ], 
2015-08-25_00:06:48.56669   "otter_facility": "otter", 
2015-08-25_00:06:48.56669   "otter_msg_type": "convergence-remove-clb-nodes", 
2015-08-25_00:06:48.56669   "otter_service": "converger", 
2015-08-25_00:06:48.56669   "scaling_group_id": "dff09d73-bcf0-4eae-972e-c4d90c6ff625", 
2015-08-25_00:06:48.56670   "tenant_id": "000000"
2015-08-25_00:06:48.56670 }
```

And the CF request-bin service:

```
2015-08-25 00:11:41+0000 [-] 

        POST http://localhost:8080/autoscale/events

        {'Content-Length': ['552'], 'Accept-Encoding': ['gzip'], 'X-Otter-Request-Id': ['b103020b-f34a-4375-ab10-1e430eeefa15'], 'Accept': ['application/json'], 'X-Auth-Token': ['token_5552e416-2973-4cca-b01f-69d5373d28dd'], 'Host': ['localhost:8080'], 'User-Agent': ['OtterScale/0.0'], 'Content-Type': ['application/vnd.rackspace.atom+json']}

        {}

        {
          "entry": {
            "content": {
              "event": {
                "product": {
                  "scalingGroupId": "dff09d73-bcf0-4eae-972e-c4d90c6ff625", 
                  "message": "Removing nodes from CLB 26924: ['144480', '852956', '972535']", 
                  "version": "1", 
                  "serviceCode": "Autoscale", 
                  "@type": "http://docs.rackspace.com/event/autoscale"
                }, 
                "region": "ORD", 
                "eventTime": "2015-08-25T00:11:41.653050Z", 
                "tenantId": "000000", 
                "version": "2", 
                "id": "4a3b7c15-1b3b-4dbb-ba3e-40fc51db9900", 
                "type": "INFO", 
                "@type": "http://docs.rackspace.com/core/event"
              }
            }, 
            "@type": "http://www.w3.org/2005/Atom", 
            "title": "autoscale"
          }
        }
```

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

File: otter/log/formatters.py [hunk 1]
```diff
@@ -30,6 +30,61 @@
 NON_PEP3101_SYSTEMS = ('kazoo',)
 
 
+_fanout = None
+
+
+def add_to_fanout(observer):
+    """
+    :return: the global instance of :class:`FanoutObserver`
```
Comment (radix) [inline otter/log/formatters.py]: I'm kinda weirded out that this returns the observer -- especially reading the use of it in setup.py. The thing that makes it weird is that if you look at the way it's used in setup.py and then try to use it in a similar way elsewhere you'll end up putting the same observer in multiple places which could be very weird. I think to emphasize that it's mutating a global observer I'd prefer to just have this return None and then require the user to explicitly call `get_fanout` when they want to add the observer.

However this is a minor nitpick so it's not really important.

**Comment by radix** [comment]
LGTM! Just one optional comment :)

File: otter/log/formatters.py (hunk 1 above)
Comment (cyli) [inline otter/log/formatters.py]: Good point!  I was mostly trying to preserve the call chain somewhat in the formatters, but hadn't really considered the usage elsewhere.  Thanks! :)

**Pull request merged**